src/test/unit/test_tracer.py

```diff
@@ -9,3 +19,83 @@ def test_access_trace_provider(self):
 
         self.assertTrue(hasattr(tracer_provider, "force_flush"))
         self.assertTrue(hasattr(tracer_provider, "shutdown"))
+
+
+class TestDependencyReport(TestCase):
+    @httpretty.activate(allow_net_connect=False)
+    @patch.dict(
+        environ, {"LUMIGO_TRACER_TOKEN": "", "LUMIGO_REPORT_DEPENDENCIES": "true"}
+    )
+    def test_dependency_report_disabled_if_no_lumigo_token(self):
+        assert not environ.get("LUMIGO_TRACER_TOKEN")
+
+        init()
+
+        assert not httpretty.latest_requests()
+
+    @httpretty.activate(allow_net_connect=False)
+    @patch.dict(
+        environ,
+        {"LUMIGO_TRACER_TOKEN": "abcdef", "LUMIGO_REPORT_DEPENDENCIES": "false"},
+    )
+    def test_dependency_report_disabled_if_lumigo_report_dependencies_false(self):
+        init()
+
+        assert not httpretty.latest_requests()
+
+    @httpretty.activate(allow_net_connect=False)
+    @patch.dict(
+        environ,
+        {
+            "LUMIGO_TRACER_TOKEN": "abcdef",
+            "LUMIGO_REPORT_DEPENDENCIES": "true",
+            "LUMIGO_ENDPOINT": "https://some.url",
+        },
+    )
+    def test_dependency_report_disabled_if_lumigo_endpoint_not_default(self):
+        init()
+
+        assert not httpretty.latest_requests()
+
+    @httpretty.activate(allow_net_connect=False)
+    @patch.dict(
+        environ, {"LUMIGO_TRACER_TOKEN": "abcdef", "LUMIGO_REPORT_DEPENDENCIES": "true"}
+    )
+    def test_dependency_report_fails(self):
+        httpretty.register_uri(
+            HTTPretty.POST,
+            "https://ga-otlp.lumigo-tracer-edge.golumigo.com/v1/dependencies",
+            headers={"Authentication": "LumigoToken abcdef"},
+            status_code=500,
+        )
+
+        init()
+
+        # Check we did call the endpoint, it failed, and init() didn't
+        assert httpretty.latest_requests()
+
+    @httpretty.activate(allow_net_connect=False)
+    @patch.dict(
+        environ, {"LUMIGO_TRACER_TOKEN": "abcdef", "LUMIGO_REPORT_DEPENDENCIES": "true"}
+    )
+    def test_dependency_report_called(self):
+        httpretty.register_uri(
+            HTTPretty.POST,
+            "https://ga-otlp.lumigo-tracer-edge.golumigo.com/v1/dependencies",
+            headers={"Authentication": "LumigoToken abcdef"},
+            status_code=200,
+        )
+
+        init()
+
+        # TODO because of the autoload mechanics, that assume that init()
+        # is only called implicitly on import, here technically we execute
+        # init twice, and we check only the latest request.
```

Comment:
Why is `init` executed twice? It is being called only once in the test.

Reply:
I think it is because of

[opentelemetry-python-distro/src/lumigo_opentelemetry/__init__.py](https://github.com/lumigo-io/opentelemetry-python-distro/blob/6f310a6e1357896c86c2ba253bcdb4659ca705d1/src/lumigo_opentelemetry/__init__.py#L169)

Line 169 in [6f310a6](https://github.com/lumigo-io/opentelemetry-python-distro/commit/6f310a6e1357896c86c2ba253bcdb4659ca705d1)
 init_data = init() 
, which we need for autowrapt to work